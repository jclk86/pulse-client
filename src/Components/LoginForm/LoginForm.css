.mobile_view_bg_login_page {
  min-height: 100vh;
  background-image: url("../../images/main_bg.jpeg");
}

.container_login_form {
  margin: 0 auto;
  min-height: 100vh;
}

.LoginForm {
  padding-top: 60px;
  padding-bottom: 60px;
}

.loading_gif {
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
  width: 10%;
}

.container_LoginForm_username,
.container_LoginForm_password {
  width: 80%;
  margin: 10px auto;
}

#label_login_username,
#label_login_password {
  display: block;
  width: 100%;
  text-align: left;
  padding-bottom: 5px;
  font-size: 18px;
  font-weight: bold;
  color: black;
}

.LoginForm_password,
.LoginForm_username {
  width: 100%;
  padding: 10px;
  font-size: 21px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
  font-size: 16px;
}

.LoginForm_password:hover,
.LoginForm_username:hover {
  border: 1px solid rgb(110, 110, 110);
}

.LoginForm_password:focus,
.LoginForm_username:focus {
  outline-width: 0;
}

.LoginForm_submit_btn {
  padding: 10px;
  min-width: 200px;
  border-radius: 10px;
  background-color: white;
  font-weight: bold;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: 2px 2px 5px;
}

.LoginForm_submit_btn:hover {
  cursor: pointer;
  border: 1px solid rgb(110, 110, 110);
}

.btn_register_link {
  color: teal;
  text-decoration: none;
}

#message_redirect {
  font-weight: bold;
  color: black;
}

.container_desktop_view_login_hero {
  padding-top: 300px;
  font-size: 40px;
  margin: 0 auto;
  width: 85%;
}

.error {
  margin-top: 2px;
  color: rgb(255, 0, 0);
}

@media only screen and (min-width: 1024px) and (orientation: landscape) {
  .LoginForm {
    margin-left: 0;
    max-width: 600px;
  }
}

@media only screen and (min-width: 1024px) {
  .mobile_view_bg_login_page {
    background-image: none;
  }
  .container_login_page {
    display: flex;
    min-height: 100vh;
  }

  .container_login_form {
    min-height: 100vh;
    background-color: rgb(211, 211, 211);
    width: 30%;
    padding-top: 40px;
  }

  .container_desktop_view_login_hero {
    color: white;
    width: 80%;
    padding-top: 250px;
  }

  .bg_desktop_view_login_image {
    background: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)),
      url("../../images/japan_night.jpeg") no-repeat center center fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    padding-bottom: 50px;
    min-height: 100vh;
    overflow: hidden;
    width: 70%;
  }

  .LoginForm {
    padding-top: 60px;
    height: 100%;
    width: 100%;
    margin: 0 auto;
    padding-left: 15px;
    padding-right: 15px;
  }
}

@media only screen and (max-width: 1023px) {
  .bg_desktop_view_login_image {
    display: none;
  }
}
