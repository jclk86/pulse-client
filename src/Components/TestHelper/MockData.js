import { DateFormatter } from "../TestHelper/MockData";

export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  article_id: 1,
  date_created: new Date(),
  author: {
    username: "Pop123"
  }
};
