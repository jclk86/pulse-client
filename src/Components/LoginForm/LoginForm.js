import React, { Component } from "react";
import { Form, Button, Input } from "../Utils/Utils";
import { withRouter, NavLink } from "react-router-dom";
import "./LoginForm.css";
import AuthApiService from "../../Services/auth-api-service";
import TokenService from "../../Services/token-service";
import GeolocationApiService from "../../Services/geolocation-api-service";

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  };

  state = {
    error: null
  };

  handleSubmitJwtAuth = event => {
    event.preventDefault();
    this.setState({ error: null });
    const { username, password } = event.target;
    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        username.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        GeolocationApiService.getUserLocation().then(location => {
          const newLocation = {
            location: `${location.city}, ${location.country_name}`
          };
          GeolocationApiService.updateUserLocation(newLocation);
        });
        this.props.onLoginSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <Form className="LoginForm" onSubmit={this.handleSubmitJwtAuth}>
        <div className="container_username">
          <label htmlFor="label__username" className="label_login">
            Username
          </label>
          <Input required name="username" id="LoginForm__username" />
        </div>
        <div className="container_password">
          <label htmlFor="LoginForm__password" className="label_login">
            Password
          </label>
          <Input
            required
            name="password"
            type="password"
            id="LoginForm__password"
            autoComplete="off"
          />
        </div>
        <div className="container_login_register">
          <p className="message_redirect">
            Not a member?{" "}
            <NavLink
              to="/registration"
              role="navigation"
              className="btn_register_link"
            >
              Register here.
            </NavLink>
          </p>
        </div>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <Button type="submit">Submit</Button>
      </Form>
    );
  }
}

export default withRouter(LoginForm);
