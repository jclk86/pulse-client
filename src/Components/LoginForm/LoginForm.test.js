import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginForm from "./LoginForm";
import sinon from "sinon";
import AuthApiService from "../../Services/auth-api-service";
import { mockUser } from "../TestHelper/MockData";

describe(`LoginForm Component`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });
  });

  it("renders LoginForm without issue", () => {
    const wrapper = shallow(<LoginForm></LoginForm>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("submits login form", () => {
    const onSubmit = sinon.spy();
    const wrapper = mount(
      <BrowserRouter>
        <LoginForm onSubmit={onSubmit}></LoginForm>
      </BrowserRouter>
    );
    wrapper.find("Form").simulate("submit", {
      target: { username: mockUser.username, password: mockUser.password },
      preventDefault: () => {}
    });
  });
});
