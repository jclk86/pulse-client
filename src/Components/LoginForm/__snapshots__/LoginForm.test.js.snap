// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders LoginForm without issue 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
